src/storefront.rs

```diff
@@ -177,21 +189,46 @@ impl StorefrontInfo {
             }
         }
 
-        fn sort_hash_map(map: &HashMap<String, String>) -> Vec<(&String, &String)> {
+        fn remove_element<'a>(parent: &'a mut Element, tag: &'a str, attr: (&'_ str, &'_ str)) {
+            let mut to_remove = parent
+                .children()
+                .enumerate()
+                .filter(|(_idx, child)| {
+                    child.tag().name() == tag && child.get_attr(attr.0) == Some(attr.1)
+                })
+                .map(|(idx, _child)| idx)
+                .collect::<Vec<_>>();
+
+            to_remove.reverse();
+
+            for idx in to_remove {
+                parent.remove_child(idx);
+            }
+        }
+
+        fn sort_hash_map(map: &HashMap<String, Option<String>>) -> Vec<(&String, &Option<String>)> {
             /* Hash maps iterate in an arbitrary order. Sort them so that diffs are less cluttered and so the tests work
             reliably. */
-            let mut values = map.iter().collect::<Vec<(&String, &String)>>();
+            let mut values = map.iter().collect::<Vec<(&String, &Option<String>)>>();
```

Comment:
Is this turbofish needed?

Reply:
Only part of it. Changed to `collect::<Vec<_>>()`.

Reply:
Looks good